### PR TITLE
feat: add Astro support

### LINKED ISSUE #154: Feature request: Astro support
It would be nice to use unplugin-icons as the go-to icon solution also in astro.

Background: I am using [astro](http://astro.build) as my static site builder and I didn't manage to use icons from unplugin-icons in my astro-files.

### PR DESCRIPTION
**Pull request by stramel**
### Description

This PR adds support for Astro. 

### Linked Issues

Closes #154 
Related to https://github.com/natemoo-re/astro-icon/issues/23

### Additional context

**NOTE:** There is an outstanding issue I discovered while developing this integration. I created a ticket as well #242

### CODE REVIEW

File: examples/astro/.gitignore [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+yarn-error.log*
+pnpm-debug.log*
+
+
+# environment variables
+.env
+.env.production
+
+# macOS-specific files
+.DS_Store
```
Comment (userquin) [inline examples/astro/.gitignore:20]: add also `.idea/`

**Comment by userquin** [comment]
@stramel astro type shouldn't be included in `package.json exports`?

**Comment by stramel** [comment]
> @stramel astro types shouldn't be included in `package.json exports` entry?

Looks like you're correct. I missed that spot. I will amend the original commit so that there is a single commit showing all spots necessary for modification outside of documentation.

> EDIT: can you also keep the example minimal?

This example was the "Basic" scaffold provided by Astro using `npx create astro@latest` and selecting the "basic" option. What do you feel needs trimmed out?

**Comment by userquin** [comment]
> This example was the "Basic" scaffold provided by Astro using `npx create astro@latest` and selecting the "basic" option. What do you feel needs trimmed out?

Remove astro components to a minimal and remove readme and external info on entry point (index route), we only need to check the icon is working, we dont need those extra info.

**Comment by stramel** [comment]
> Remove astro components to a minimal and remove readme and external info on entry point (index route), we only need to check the icon is working, we dont need those extra info.

Alright, I have updated my PR with those changes.

NOTE: Just for consistency, the `next` and `nuxt3` examples should probably be updated to reflect this minimal approach as well. (These were the 2 I looked at initially for reference)

**Comment by userquin** [comment]
@stramel I found on latest Astro versions a lot of sourcemap points missing source files (also for astro pages), for example, running your astro example:
```shell
Sourcemap for "~icons/icon-park/abnormal.astro" points to missing source files
Sourcemap for "~icons/mdi/store-24-hour.astro" points to missing source files
Sourcemap for "~icons/mdi/alarm-off.astro" points to missing source files
Sourcemap for "~icons/logos/astro.astro" points to missing source files
```

**Review by userquin — APPROVED** [review]

**Comment by stramel** [comment]
> @stramel I found on latest Astro versions a lot of sourcemap points missing source files (also for astro pages), for example, running your astro example:
> 
> ```shell
> Sourcemap for "~icons/icon-park/abnormal.astro" points to missing source files
> Sourcemap for "~icons/mdi/store-24-hour.astro" points to missing source files
> Sourcemap for "~icons/mdi/alarm-off.astro" points to missing source files
> Sourcemap for "~icons/logos/astro.astro" points to missing source files
> ```

@userquin I also noticed that when running the example. I wasn't sure if that was a side-effect of the virtual components or an issue? I'm happy to dig into it but unsure where to start digging, can you point me in a direction?

**Comment by userquin** [comment]
> @userquin I also noticed that when running the example. I wasn't sure if that was a side-effect of the virtual components or an issue? I'm happy to dig into it but unsure where to start digging, can you point me in a direction?

I'm no an expert on Astro, if the sfc comes from disk or from virtual module doesn't matter, we also have the same problem with other compilers, Anthony just include an empty sourcemap here https://github.com/antfu/unplugin-icons/blob/main/src/index.ts#L50, maybe we can ask Nate.

**Comment by userquin** [comment]
> @userquin Seems like a recent bug on the Astro side. [withastro/astro#5053](https://github.com/withastro/astro/issues/5053)

Yeah, I just send a direct message to Nate, it seems there is a few reports about those warnings.

**Comment by stramel** [comment]
Rebased onto main to remove merge conflicts

**Comment by userquin** [comment]
> Rebased onto main to remove merge conflicts

Can you run `pnpm run lint` from root? Just to confirm it is working with Astro files and there are no errors...

**Comment by userquin** [comment]
@stramel I will push changes running `pnpm run lint-fix`...

**Review by userquin — APPROVED** [review]

**Pull request merged**